### ISSUE TITLE
`adapative_{avg, max}_pool` will crash in the backward pass on cuda on empty output size

### ISSUE THREAD

**Issue by upupyy** [pytorch/pytorch#78868]
### 🐛 Describe the bug

`adapative_{avg, max}_pool` will crash in the backward pass on cuda

```python
import torch

input = torch.rand([1, 64, 10, 9, 8], dtype=torch.float32, device='cuda', requires_grad=True)
output_size = 0

res = torch.nn.functional.adaptive_avg_pool3d(input, output_size)
print('forward')

res.sum().backward()
# Crash
print('backward')
```

It will try to compute the gradient based on an empty tensor.

### Versions

pytorch: 1.11.0
cuda: 11.3

cc @ezyang @gchanan @zou3519 @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are @ngimel

**Comment by jbschlosser** [comment]
We'd accept a PR guarding against 0 `output_size`.

**Comment by anshradh** [comment]
Hello! I'd like to pick this up if no one has already.

**Comment by anshradh** [comment]
[#pull/81171](https://github.com/pytorch/pytorch/pull/81171)

**Comment by anshradh** [comment]
https://github.com/pytorch/pytorch/pull/81171
